sdk/src/extras/disk.rs

```diff
@@ -796,6 +770,64 @@ pub mod types {
         pub image_version: String,
     }
 
+    struct UploadWorker {
+        client: Client,
+        disk: Name,
+        project: NameOrId,
+        file: File,
+        file_path: PathBuf,
+        progress_tx: watch::Sender<u64>,
+        upload_progress: Arc<AtomicU64>,
+        buf: Vec<u8>,
+    }
+
+    impl UploadWorker {
+        async fn upload_chunk(&mut self, offset: u64) -> Result<(), DiskImportError> {
+            self.file.seek(SeekFrom::Start(offset))?;
+            let n = match self
+                .file
+                .by_ref()
+                .take(CHUNK_SIZE)
+                .read_to_end(&mut self.buf)
```

Comment:
If I understand this correctly, this is blocking. I think this was less of a problem when we had a task doing the reading.